src/librbd/AsioEngine.cc

```diff
@@ -31,7 +30,9 @@ AsioEngine::AsioEngine(std::shared_ptr<librados::Rados> rados)
   if (rbd_threads > rados_threads) {
     // inherit the librados thread count -- but increase it if librbd wants to
     // utilize more threads
-    m_cct->_conf.set_val("librados_thread_count", stringify(rbd_threads));
+    m_cct->_conf.set_val_or_die("librados_thread_count",
+                                std::to_string(rbd_threads));
+    m_cct->_conf.apply_changes(nullptr);
```

Comment:
Nice catch. I see another occurrence of set_val(), but no apply_changes in `librbd/api/Mirror.cc`. Is that needed there too or is above a special case?

Reply:
The above is a special case because the value is getting reset on a live RADOS instance.  In the case you are referring to
https://github.com/ceph/ceph/blob/9f0440e6d5f702ad36d7d44ccf2a69018a2dd653/src/librbd/api/Mirror.cc#L1409-L1417
the initial values are set (before `connect` is even called).

Reply:
Okay understood.